enumerable_methods.rb

```diff
@@ -0,0 +1,106 @@
+# Project 2: Enumerable Methods
+module Enumerable
+  def my_each(&block)
+    size.times { |i| block.call(self[i]) } and return self if block
+    to_enum
+  end
+
+  def my_each_with_index(&block)
+    my_each { |idx| block.call(self[idx], idx) } and return self if block_given?
+    to_enum
+  end
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`each_with_index`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-each_with_index). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_each_with_index.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] calls the given block once for each element in self